### PR TITLE
MULE-11589: Add support for plugins to the MuleController and MuleDep…

### PR DESCRIPTION
**Pull request by anosenzo**
…loyment Test Classes

### CODE REVIEW

File: tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java [hunk 1]
```diff
@@ -68,7 +71,8 @@ public Controller(String muleHome, int timeout) {
     this.muleBin = getMuleBin();
     this.domainsDir = new File(muleHome + "/domains");
     this.appsDir = new File(muleHome + "/apps/");
-    this.libsDir = new File(muleHome + "/lib/user");
+    this.libsDir = new File(muleHome + LIB_USER_FOLDER);
+    this.pluginsDir = new File(muleHome + PLUGINS_FOLDER);
```
Comment (aiannucci) [inline tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java]: What is the purpose of this change if plugins live inside applications and not in muleHome/plugins?

File: tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java [hunk 2]
```diff
@@ -159,18 +163,28 @@ protected void deployDomain(String domain) {
...
 
-  protected void addLibrary(File jar) {
-    verify(jar.exists(), "Jar file does not exist: %s", jar);
-    verify("jar".equals(FilenameUtils.getExtension(jar.getAbsolutePath())), "Library [%s] don't have .jar extension.", jar);
-    verify(jar.canRead(), "Cannot read jar file: %s", jar);
-    verify(libsDir.canWrite(), "Cannot write on lib dir: %", libsDir);
+  protected void addFileToFolder(File file, String fileExtension, File folderToCopyFile, String artifactType,
+                                 String errorMessage) {
+    verify(file.exists(), "%s file does not exist: %s", artifactType, file);
+    verify(fileExtension.equals(FilenameUtils.getExtension(file.getAbsolutePath())), "[%s] [%s] don't have .[%s] extension.",
```
Comment (aiannucci) [inline tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java]: don't -> doesn't

File: tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java [hunk 3]
```diff
@@ -159,18 +163,28 @@ protected void deployDomain(String domain) {
     }
   }
 
-  protected void addLibrary(File jar) {
-    verify(jar.exists(), "Jar file does not exist: %s", jar);
-    verify("jar".equals(FilenameUtils.getExtension(jar.getAbsolutePath())), "Library [%s] don't have .jar extension.", jar);
-    verify(jar.canRead(), "Cannot read jar file: %s", jar);
-    verify(libsDir.canWrite(), "Cannot write on lib dir: %", libsDir);
+  protected void addFileToFolder(File file, String fileExtension, File folderToCopyFile, String artifactType,
```
Comment (aiannucci) [inline tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java]: Is this meant to be called externally? If not, it should be private.

File: tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java (hunk 1 above)
Comment (anosenzo) [inline tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java]: The compatibility Module should go at App level too??
I follow the logic in the compatibility test done by @elrodro83 https://github.com/mulesoft/mule-ee/blob/mule-4.x/distributions/tests/src/test/java/com/mulesoft/mule/distributions/server/CompatibilityAppControlTestCase.java#L60
